### PR TITLE
[ffmpeg] Update dependency support for recent changes.

### PR DESCRIPTION
**Pull request by Sibras**
Updates for changes in commits:
7bb175e [aom/libavif] Add support for ARM and UWP
a8204d9 [fribidi] Support new platform

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your PR!

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I think these messages should just be deleted; they are implied by the dependencies and vcpkg will already print such messages long before attempting to build ffmpeg:

```
PS C:\Dev\vcpkg> .\vcpkg.exe install ffmpeg[aom]:arm-windows
Computing installation plan...
Error: aom[core] is only supported on '!(windows & arm & !uwp)'
PS C:\Dev\vcpkg>
```

Similarly, I believe supports expressions on features that exist only to accommodate dependencies should be removed.

Thanks for noticing the duplicate message!

**Comment by Sibras** [comment]
The supports 'platform' values in vcpkg.json are there to enable the 'all' feature to work correctly and the error messages were added to the portfile because at the time vcpkg wasn't generating error messages when unsupported features were enabled (#17153) and im not entirely sure if that situation has been resolved.

I should also point out that some of the errors in the portfile and not just due to the dependencies support requirements but also additional requirements of ffmpeg (i.e. vcpkg may be able to build a dependency with configuration x but ffmpeg doesnt support using it)

File: ports/ffmpeg/portfile.cmake [hunk 1]
```diff
@@ -17,8 +17,8 @@ if ("alsa" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_LINUX)
 endif()
 
 if("aom" IN_LIST FEATURES)
-    if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64" OR VCPKG_TARGET_IS_UWP)
-        message(FATAL_ERROR "Feature 'aom' does not support 'uwp | arm'")
+    if ((VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64") AND NOT VCPKG_TARGET_IS_UWP)
+        message(FATAL_ERROR "Feature 'aom' does not support 'windows & arm & !uwp'")
     endif()
 endif()
```
Comment (PhoebeHui) [inline ports/ffmpeg/portfile.cmake:23]: I think Billy means these messages should just be deleted, it looks doesn't support arm or arm64, which the dependency port aom already have the message.

```
PS F:\vcpkg\21980\vcpkg> ./vcpkg install ffmpeg[aom]:arm-windows
Computing installation plan...
Error: aom[core] is only supported on '!(windows & arm & !uwp)'
```

**Comment by BillyONeal** [comment]
> The supports 'platform' values in vcpkg.json are there to enable the 'all' feature to work correctly

That's not correct; feature resolution does not inspect `"supports"`

> I should also point out that some of the errors in the portfile and not just due to the dependencies support requirements but also additional requirements of ffmpeg (i.e. vcpkg may be able to build a dependency with configuration x but ffmpeg doesnt support using it)

In such cases we can leave the message :). I just don't want `ffmpeg` to need to be updated every time  whether one of its dependencies supports a given platform changes since it will inevitably create out-of-date messages.

**Comment by Sibras** [comment]
> > The supports 'platform' values in vcpkg.json are there to enable the 'all' feature to work correctly
> 
> That's not correct; feature resolution does not inspect `"supports"`

No, but it does check "platform" which is what I was referring to

> > I should also point out that some of the errors in the portfile and not just due to the dependencies support requirements but also additional requirements of ffmpeg (i.e. vcpkg may be able to build a dependency with configuration x but ffmpeg doesnt support using it)
> 
> In such cases we can leave the message :). I just don't want `ffmpeg` to need to be updated every time whether one of its dependencies supports a given platform changes since it will inevitably create out-of-date messages.

I can go through and delete everything thats not ffmpeg specific. I wasnt a big fan of adding that in the first place for just the reasons you mentioned.

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updated reporting and working with us on code review feedback!

**Pull request merged**